### PR TITLE
ci: set GHA for updating flake.lock

### PR DESCRIPTION
**Pull request by detroyejr**
Fixes https://github.com/pola-rs/r-polars/issues/1231#issuecomment-2351481317

The nix flake lockfile needed an update to support the version of rust r-polars uses. 

Nix build should work after this.

### CODE REVIEW

**Comment by eitsupi** [comment]
Thanks for the quick update!

Do you know if we can set up CI for updating nix flake automatically?

**Comment by detroyejr** [comment]
Looks like that's possible.

Setup an action that can run on a scheduled basis. Opens up a pr that looks like [this](https://github.com/detroyejr/r-polars/pull/1).

File: .github/workflows/flake.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: update-flake-lock
+on:
+  workflow_dispatch: # allows manual triggering
+  schedule:
+    - cron: '0 0 * * 0' # runs weekly on Sunday at 00:00
```
Comment (detroyejr) [inline .github/workflows/flake.yml:5]: If this is something you want to incorporate, let me know if you want to change the default.

**Review by eitsupi — APPROVED** [review]
Thanks, this looks good!
Let's merge and see what happens.

File: .github/workflows/flake.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (eitsupi) [inline .github/workflows/flake.yml:6]:
```suggestion
permissions:
  contents: write

```

**Comment by detroyejr** [comment]
@eitsupi Just a quick note on this, it looks like the scheduled run didn't quite work. If I'm not mistaken, the token provisioned for actions can push to [branches](https://github.com/pola-rs/r-polars/tree/update_flake_lock_action), but [isn't able to open PR's](https://github.com/pola-rs/r-polars/actions/runs/10976693004/job/30477726181#step:4:416).

Everything else on the run looked okay so I think this will go be good to go when that's updated.

**Pull request merged**